packages/react-reconciler/src/ReactUpdateQueue.js

```diff
@@ -526,20 +502,16 @@ export function processUpdateQueue<State>(
   if (newFirstUpdate === null) {
     queue.lastUpdate = null;
   }
-  if (newFirstCapturedUpdate === null) {
-    queue.lastCapturedUpdate = null;
-  } else {
-    workInProgress.effectTag |= Callback;
-  }
-  if (newFirstUpdate === null && newFirstCapturedUpdate === null) {
+  queue.lastCapturedUpdate = null;
+  if (newFirstUpdate === null) {
     // We processed every update, without skipping. That means the new base
     // state is the same as the result state.
     newBaseState = resultState;
   }
 
   queue.baseState = newBaseState;
   queue.firstUpdate = newFirstUpdate;
-  queue.firstCapturedUpdate = newFirstCapturedUpdate;
+  queue.firstCapturedUpdate = null;
 
```

Comment:
I don't know why even I remove this line i.e. `queue.firstCapturedUpdate = newFirstCapturedUpdate`, it still works, maybe the queue has been set to `null` after we executed this function? I don't know, don't dig into it. So I still remain this line